ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcCommandCompletionService.scala

```diff
@@ -21,26 +21,44 @@ import scalaz.syntax.tag._
 
 import scala.concurrent.Future
 
+object GrpcCommandCompletionService {
+
+  private[this] val completionStreamDefaultOffset =
+    LedgerOffset.of(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_END))
+
+  private def fillInWithDefaults(request: CompletionStreamRequest): CompletionStreamRequest = {
+    if (request.offset.isDefined) {
+      request
+    } else {
+      request.withOffset(completionStreamDefaultOffset)
+    }
+  }
+
+}
+
 class GrpcCommandCompletionService(
     ledgerId: LedgerId,
     service: CommandCompletionService,
     partyNameChecker: PartyNameChecker
 )(implicit protected val esf: ExecutionSequencerFactory, protected val mat: Materializer)
     extends CommandCompletionServiceAkkaGrpc {
 
+  import GrpcCommandCompletionService.fillInWithDefaults
+
   private val validator = new CompletionServiceRequestValidator(ledgerId, partyNameChecker)
 
   override def completionStreamSource(
-      request: CompletionStreamRequest): Source[CompletionStreamResponse, akka.NotUsed] =
+      request: CompletionStreamRequest): Source[CompletionStreamResponse, akka.NotUsed] = {
     validator
-      .validateCompletionStreamRequest(request)
+      .validateCompletionStreamRequest(fillInWithDefaults(request))
```

Comment:
In other services/validators we followed the approach that the domain object closely resembles the semantics of the ledger api in terms of optional/required fields. This means that the domain `CompletionStreamRequest` should get an optional `end: Option[LedgerOffset]` field, that is then filled in with the LedgerEnd logic in the implementation of the `domain.CommandCompletionService`. You can look at a similarly shaped pattern for the `GrpcTransactionService`, `TransactionServiceRequestValidator`, and `GetTransactionsRequest#end`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/1961/commits/1f8fae9ab62c717e1424427319070ee4e186564f